Lib/test/test_typing.py

```diff
@@ -162,13 +170,33 @@ class NoReturnTests(BottomTypeTestsMixin, BaseTestCase):
     def test_repr(self):
         self.assertEqual(repr(NoReturn), 'typing.NoReturn')
 
+    def test_get_type_hints(self):
+        def some(arg: NoReturn) -> NoReturn: ...
+
+        self.assertEqual(
+            get_type_hints(some),
+            {'arg': NoReturn, 'return': NoReturn},
+        )
```

Comment:
A version of this with a stringized `NoReturn` tests the `_type_check` codepath for `NoReturn`. Ditto for the other cases. An Annotated version will test the same thing.

Reply:
Do you propose adding `'NoReturn'` and other stringified cases? Am I right?

Reply:
```python
        def some(arg: 'NoReturn') -> NoReturn: ...

        self.assertEqual(
            get_type_hints(some),
            {'arg': NoReturn, 'return': NoReturn},
        )
```

if `NoReturn` is removed from this line: https://github.com/python/cpython/blob/06e1701ad3956352bc0f42b8f51c2f8cc85bf378/Lib/typing.py#L180

The above case fails, but not with your listed case.

Reply:
The relevant bit is you're testing `typing._type_check` getting called on these typeforms. It's not called normally, but it will get called if you stringize it and do `get_type_hints` (as part of the resolution of the stringification). It would also get called if you did e.g. `Annotated[NoReturn, ""]`, because `Annotated` calls it.

Reply:
Done, great addition! 👍